### PR TITLE
Fix Various Import/Export issues

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This splits off the fixes from #11228 into a separate PR so that I don't need to remake all the replays immediately.

### CODE REVIEW

File: src/openrct2/rct2/S6Exporter.cpp [hunk 1]
```diff
@@ -1031,9 +1031,10 @@ void S6Exporter::ExportSpriteVehicle(RCT2SpriteVehicle* dst, const Vehicle* src)
     }
     else
     {
+        // Track direction and type are in the same field
         dst->track_direction = src->track_direction;
+        // dst->track_type = src->track_type;
```
Comment (tupaschoal) [inline src/openrct2/rct2/S6Exporter.cpp:1036]: I found it a bit weird that it doesn't export to `track_type` anymore, but it does read from it when importing.

Comment (janisozaur) [reply src/openrct2/rct2/S6Exporter.cpp:1036]: https://github.com/OpenRCT2/OpenRCT2/blob/0c6828f69204a539d497e2b80c642dff95bd4440/src/openrct2/rct2/S6Exporter.cpp#L1034
https://github.com/OpenRCT2/OpenRCT2/blob/0c6828f69204a539d497e2b80c642dff95bd4440/src/openrct2/ride/Vehicle.h#L198-L202

Comment (janisozaur) [reply src/openrct2/rct2/S6Exporter.cpp:1036]: The code is full of such optimisations

Comment (tupaschoal) [reply src/openrct2/rct2/S6Exporter.cpp:1036]: I see.

Comment (duncanspumpkin) [reply src/openrct2/rct2/S6Exporter.cpp:1036]: And also boat location user the same field in sv6. So before it was writing boat location info into the track type.

**Pull request merged**